### PR TITLE
 Add Maxine BNR 2.0 network implementation to NeMo

### PR DESCRIPTION
**Pull request by nitin9252**
# What does this PR do ?

Add Maxine BNR 2.0 network implementation to NeMo

**Collection**: Audio

# Changelog

- Add initial implementation of network + training script/tutorial

# Usage

- You can potentially add a usage example below
Please refer to added training script/tutorial

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: examples/audio/audio_to_audio_train.py [hunk 1]
```diff
@@ -39,6 +39,8 @@
     SchroedingerBridgeAudioToAudioModel,
     ScoreBasedGenerativeAudioToAudioModel,
 )
+from nemo.collections.audio.models.maxine import BNR2
+
```
Comment (nasretdinovr) [inline examples/audio/audio_to_audio_train.py:43]: Is newline necessary here?

File: nemo/collections/audio/losses/maxine/losses_combined.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+from typing import Optional
+from torchaudio.transforms import MelSpectrogram
+from torchaudio.functional import resample
+
+import hashlib
+from pathlib import Path
+from nemo.collections.asr.models import ASRModel
+from nemo.utils.cloud import maybe_download_from_cloud
+from nemo.utils.data_utils import resolve_cache_dir
+from nemo.utils import logging
```
Comment (nasretdinovr) [inline nemo/collections/audio/losses/maxine/losses_combined.py:29]: I think that imports aren't sorted properly according to PEP8 standards

**Comment by chtruong814** [comment]
@anteju @nitin9252 

I believe currently we'll need to import guard torchaudio. Here's how I see it currently being imported in the code and tests for example.  
https://github.com/NVIDIA/NeMo/blob/9d4e41e5d73a7ffeabb00a1f36982716b306dd93/nemo/collections/audio/modules/transforms.py#L24

https://github.com/NVIDIA/NeMo/blob/9d4e41e5d73a7ffeabb00a1f36982716b306dd93/tests/collections/audio/test_audio_transforms.py#L195

**Comment by chtruong814** [comment]
Canceling the CI runs until after the updates are done to add the torchaudio import guards.
